src/librustc_codegen_ssa/mir/place.rs

```diff
@@ -454,16 +455,24 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             mir::PlaceRef {
                 base: mir::PlaceBase::Static(box mir::Static {
                     ty,
-                    kind: mir::StaticKind::Promoted(promoted),
+                    kind: mir::StaticKind::Promoted(promoted, substs),
+                    def_id,
                 }),
                 projection: None,
             } => {
                 let param_env = ty::ParamEnv::reveal_all();
+                let instance = Instance::new(*def_id, substs.subst(bx.tcx(), self.instance.substs));
+                debug!("instance: {:?}", instance);
                 let cid = mir::interpret::GlobalId {
-                    instance: self.instance,
+                    instance: instance,
                     promoted: Some(*promoted),
                 };
-                let layout = cx.layout_of(self.monomorphize(&ty));
+                let mono_ty = tcx.subst_and_normalize_erasing_regions(
```

Comment:
we should have already brought this into this mir::Body's universe during inlining. I don't think we need to change anything compared to the original code.

Reply:
So, if I remove this, I get the same ICE as above because `instance.substs` != `self.instance.substs` which is what `self.monomorphize()` uses.

Reply:
`instance.substs != self.instance.substs` can only happen when inlining, right? Are test with mir optimizations at level 2 failing? or just arbitrary tests? If it's the latter, something is very odd. Try dumping both instances if they differ.